### ISSUE TITLE
Parent selector should be considered when evaluating valid pseudoclasses

### ISSUE THREAD

**Issue by jathak** [stylelint-scss/stylelint-scss#596]
This Sass code:
```scss
::-webkit-scrollbar {
  &:horizontal {
    border-top: 1px;
  }
}
```
currently results in a `selector-pseudo-class-no-unknown` lint, but without the nesting
```scss
::-webkit-scrollbar:horizontal {
  border-top: 1px;
}
```
there's no lint, since Stylelint correctly knows that `:horizontal` is a valid pseudoselector when applied to the `::-webkit-scrollbar` pseudoelement.

**Comment by kristerkari** [comment]
@jathak the rule that you are referring to is from the https://github.com/stylelint/stylelint repo.

Could you please report the problem there instead?
